cpp/source/deal.II/new_supercapacitor.templates.h

```diff
@@ -0,0 +1,358 @@
+  unsigned int const electrochemical_block      = database.get<unsigned int>("electrochemical_block");
+  unsigned int const n_blocks                   = database.get<unsigned int>("n_blocks");
+  // clang-format on
+  // For now, we have just one block so we can ignore some variable.
+  std::ignore = temperature_component;
+  std::ignore = solid_potential_component;
+  std::ignore = liquid_potential_component;
+  std::ignore = thermal_block;
+  std::ignore = electrochemical_block;
+  std::ignore = n_blocks;
+
+  // read material properties
+  std::shared_ptr<boost::property_tree::ptree> material_properties_database =
+      std::make_shared<boost::property_tree::ptree>(
+          database.get_child("material_properties"));
+  MPValuesParameters<dim> params(material_properties_database);
+  params.geometry = geometry;
+  std::shared_ptr<MPValues<dim>> mp_values =
+      std::make_shared<MPValues<dim>>(params);
+
+  std::shared_ptr<boost::property_tree::ptree> boundary_values_database =
+      std::make_shared<boost::property_tree::ptree>(
+          database.get_child("boundary_values"));
+  std::shared_ptr<SuperCapacitorBoundaryValues<dim>> boundary_values =
+      std::make_shared<SuperCapacitorBoundaryValues<dim>>(
+          SuperCapacitorBoundaryValuesParameters<dim>(
+              boundary_values_database));
+
+  // Initialize the electrochemical physics parameters
+  electrochemical_physics_params.reset(
+      new ElectrochemicalPhysicsParameters<dim>(
+          std::make_shared<boost::property_tree::ptree>(database)));
+  electrochemical_physics_params->dof_handler = dof_handler;
+  electrochemical_physics_params->mp_values =
+      std::dynamic_pointer_cast<MPValues<dim> const>(mp_values);
+  electrochemical_physics_params->boundary_values =
+      std::dynamic_pointer_cast<BoundaryValues<dim> const>(boundary_values);
+
+  // Compute the surface area. This is neeeded by several evolve_one_time_step_*
+  surface_area = 0.;
+  dealii::types::boundary_id cathode_boundary_id =
+      database.get<dealii::types::boundary_id>(
+          "boundary_values.cathode_boundary_id");
+  dealii::QGauss<dim - 1> face_quadrature_rule(fe->degree + 1);
+  unsigned int const n_face_q_points = face_quadrature_rule.size();
+  dealii::FEFaceValues<dim> fe_face_values(*fe, face_quadrature_rule,
+                                           dealii::update_JxW_values);
+  for (auto cell : dof_handler->active_cell_iterators())
+    if (cell->at_boundary())
+      for (unsigned int face = 0;
+           face < dealii::GeometryInfo<dim>::faces_per_cell; ++face)
+        if ((cell->face(face)->at_boundary()) &&
+            (cell->face(face)->boundary_id() == cathode_boundary_id))
+          for (unsigned int face_q_point = 0; face_q_point < n_face_q_points;
+               ++face_q_point)
+          {
+            fe_face_values.reinit(cell, face);
+            surface_area += fe_face_values.JxW(face_q_point);
+          }
```

Comment:
I agree that it is bad place for it but it makes no sense for the post processor to compute data that is needed for the computation. I could move it to the geometry but (1) I didn't want to touch too much of the code in once (2) if the surface is curved, you need a ```dof_handler``` to exist to compute the area.